code/handlers/ldap.q

```diff
@@ -45,34 +69,38 @@ login:{[user;pass]                                              / validate login
       update attempts:0, blocked:0b from `.ldap.cache where user=user];
   ];
 
+  /authenticate:{[dict].ldap.bind[.ldap.sessionID;(enlist `$dict[`bind_dn])!enlist `$dict[`pass]]}
+
   authorised:$[all (                                            / check if previously used details match
     incache[`success];                                          / previous attempt was a success
     incache[`time]>.z.p-.ldap.checktime;                        / previous attemp occured within checktime period
     incache[`pass]~np:md5 pass                                  / same password was used
   );
     1b;
-    @[{.ldap.authenticate[x]`success};dict;0b]                  / attempt authentication
+    authenticate:{[dict].ldap.bind[.ldap.sessionID;(`dn;`cred)!(dict[`bind_dn];`$dict[`pass])]} [dict]                 / attempt authentication
```

Comment:
mentioned above, but this becomes simplified. also don't think you need the lambda, and I don't think you do anything with the stored `authenticate` variable?

```suggestion
    .ldap.bind[.ldap.sessionID;dict]                 / attempt authentication
```

Before we have this in an error trap, probably still want it in an error trap.